doc/port-help.1
'\" t
.TH "PORT\-HELP" "1" "2\&.12\&.99" "MacPorts 2\&.12\&.99" "MacPorts Manual"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
port-help \- Get help on MacPorts commands
.SH "SYNOPSIS"
.sp
.nf
\fBport\fR \fBhelp\fR [\fIaction\fR | \fIhelp\-topic\fR]
.fi
.SH "DESCRIPTION"
.sp
\fBport help\fR opens the manpage for a given port action such as \fIinstall\fR, \fIupgrade\fR, \fIselfupdate\fR, etc\&. or a specific help topic like \fImacports\&.conf\fR\&. MacPorts even has help for its help feature\&. MacPorts will however not engage in higher level abstractions: There is no help on the help on the help\&.
.sp
Further documentation on MacPorts can be found in the following places:
.PP
\m[blue]\fBThe MacPorts Guide\fR\m[]\&\s-2\u[1]\d\s+2
.RS 4
Detailed help and support guide
.RE
.PP
\m[blue]\fBThe MacPorts Wiki\fR\m[]\&\s-2\u[2]\d\s+2
.RS 4
MacPorts usage, development, and support
.RE
.PP
\m[blue]\fBMailing Lists\fR\m[]\&\s-2\u[3]\d\s+2
.RS 4
Ask questions on the mailing lists: macports\-dev for base and Portfile development, macports\-users for user support questions
.RE
.PP
\m[blue]\fBFAQ\fR\m[]\&\s-2\u[4]\d\s+2
.RS 4
Frequently Asked Questions
.RE
.PP
\m[blue]\fBProblem Hotlist\fR\m[]\&\s-2\u[5]\d\s+2
.RS 4
Solutions for the most common error messages and other problems
.RE
.SH "SEE ALSO"
.sp
\fBport\fR(1)
.SH "AUTHORS"
.sp
.if n \{\
.RS 4
.\}
.nf
(C) 2014 The MacPorts Project
Clemens Lang <cal@macports\&.org>
.fi
.if n \{\
.RE
.\}
.SH "EXTERNAL REFERENCES"
.IP " 1." 4
The MacPorts Guide
.RS 4
\%https://guide.macports.org/
.RE
.IP " 2." 4
The MacPorts Wiki
.RS 4
\%https://trac.macports.org/wiki/
.RE
.IP " 3." 4
Mailing Lists
.RS 4
\%https://trac.macports.org/wiki/MailingLists
.RE
.IP " 4." 4
FAQ
.RS 4
\%https://trac.macports.org/wiki/FAQ
.RE
.IP " 5." 4
Problem Hotlist
.RS 4
\%https://trac.macports.org/wiki/ProblemHotlist
.RE


NEWS
== Overview of changes from MacPorts 2.0.4 to 2.1.0 ==

# For regular users:

*) New 'rev-upgrade' action that checks for and rebuilds ports that may have
   become incorrectly linked when a dependency was upgraded to a new,
   binary-incompatible version. This runs automatically after upgrades and
   installs by default, and can be turned off or configured to only report
   problems in macports.conf.

*) The pkg and mpkg targets will create flat packages on 10.6+.

*) Build-time dependencies will no longer be installed if a prebuilt
   archive is available for download (and port is configured to use it).

*) Config file archive_sites.conf can be used to download archives from
   custom locations.

*) 'buildfromsource' macports.conf option allows permanently choosing
   source-only or binary-only mode.

*) New macports.conf options 'host_blacklist' and 'preferred_hosts', for
   controlling which download hosts will be used.

# For Portfile writers:

*) New -W option for reinplace that sets a base directory for all file
   arguments.

*) Livecheck can now find multiple regex matches on each line.

*) license_noconflict option can be used to override automatic license
   compatibility checks.

*) compiler.whitelist, compiler.blacklist, and compiler.fallback options allow
   simpler, detailed control over which compilers are used to build ports.


== Overview of changes from MacPorts 1.9.2 to 2.0.0 ==

# For regular users:

*) Downloaded archives are now checked for integrity using digital signatures.
   New pubkeys.conf file allows configuring keys to trust.

*) Actions can be abbreviated to an unambiguous prefix of the full name.

*) Ports that create StartupItems (aka launchd plists) should now be
   installable by unprivileged users.

*) Port images are now stored as archives. Archive mode is now effectively
   always on. Direct mode and the flat registry format are no longer
   available. Conversion will happen automatically.

# For Portfile writers:

*) Multiple ports can now be defined in one Portfile using the 'subport'
   procedure.

*) New option 'add_users' can be set to ensure that the listed users are
   created at the right times during the install process.

*) SHA256 checksums are now supported.

== Overview of changes from MacPorts 1.8.2 to 1.9.0 ==

# For regular users:

*) All output is now logged to a file, whose location will be shown if a port
   target fails. This should be attached to bug reports. The new 'log' action
   can also be used to view the log for a port.
*) Command line options can be abbreviated if unambiguous.
*) New actions: rdeps, rdependents, setrequested, unsetrequested
*) --follow-dependencies option for uninstall (ports with the requested flag
   set or with other dependents are skipped).
*) New pseudo-ports and selectors: requested, unrequested, leaves, depof:,
   rdepof:, dependentof:, rdependentof:.
*) Architectures of dependencies are checked before install, and in many cases
   will be automatically fixed by upgrading with +universal.
*) New sqlite registry format which is faster and lacks some of the old
   format's inconvenient limitations.
*) Explicitly deselected variants (e.g. -foo) are now recorded, so they will
   not be set to + when upgrading if +foo appears in variants.conf or
   the port's default_variants.

# For Portfile writers:

*) New os.subplatform variable.
*) New configure.ld_archflags option to match configure.cc_archflags et al.
*) 'platform' acts like an 'if' statement rather than a variant.
*) New supported_archs option for indicating a port only supports certain
   archs or is noarch.
*) pre-/post-deactivate and pre-/post-uninstall procedures can now be used,
   and pre-/post-activate procedures are run every time the port is activated,
   not just the first time when it is installed.

== Overview of changes from MacPorts 1.7.1 to 1.8.0 ==

# For regular users:

*) Compatible with Mac OS X 10.6 Snow Leopard, requires at least 10.4 Tiger
*) CPU architecture to build for is now configurable
*) Performance and memory usage improvements
*) New -y option for dry-run
*) 'port search' has many new options that make it more flexible
*) 'port upgrade' now follows the right dependencies when variants change them
*) 'port info' and 'port variants' indicate selected and default variants;
   'port variants' also shows conflicts and dependencies between variants
*) A port can be marked as being replaced by another port; upgrading will
   perform the replacement
*) upgrade accepts new options: --force, --enforce-variants, --no-replace
*) Ports can be marked as conflicting, and conflicts will be detected before
   the install process starts
*) Proxy information can now be obtained from System Preferences and from
   macports.conf
*) Activating a port will now first deactivate any other version of the same
   port that is already active
*) MacPorts can be configured to drop root privileges when not needed
*) port(1) will exit with an appropriate status code when an error is
   encountered (use -p to continue despite errors)

# For Portfile writers:

*) New options: replaced_by, conflicts, depends_fetch, depends_extract, license
*) build_arch variable indicates which architecture should be targeted. Flags
   used to implement this are configured per-compiler, for example,
   configure.cc_archflags will be passed to configure.cc in CFLAGS
*) Dependencies can be satisfied by ports that are installed but not
   available in the ports tree
*) Deleting all the contents of an option no longer unsets the option entirely
*) Using an alternative fetch or extract type now adds appropriate
   dependencies automatically, e.g. 'use_zip yes' adds a dep on unzip, and
   'fetch.type svn' adds a dep on subversion
*) Running phases prior to install should now work reliably when the port is
   already installed

== Overview of changes from MacPorts 1.6.0 to 1.7.0 ==

# For regular users:


src/port/port-help.tcl
# This file contains help strings for the various commands/topics in port(1)
#
# Many of these strings are place-holders right now.  Replace with genuinely
# helpful text and then delete this message.
#
# port-help.tcl
# $Id$


set porthelp(activate) {
Activate the given ports
}

set porthelp(archive) {
Archive the given ports
}

set porthelp(build) {
Build the given ports
}

set porthelp(cat) {
Writes the Portfiles of the given ports to stdout
}

set porthelp(cd) {
Changes to the directory of the given port

Only in interactive mode.
}

set porthelp(checksum) {
Compares the checksums for the downloaded files of the given ports
}

set porthelp(clean) {
Removes file associates with given ports

--archive     Removes created archives
--dist        Removes downloaded distfiles
--work        Removes work directory (default)
--all         Removes everything from above
--logs 				Removes log files for port
}

set porthelp(log) {
Shows main log for given ports

--phase <phase>		Filteres log file by phase (configure, fetch, etc..)
--verbosity <ver>	Filteres log file by verbosity level (msg, debug, info)	
}

set porthelp(configure) {
Configure the given ports
}

set porthelp(contents) {
Returns a list of files installed by given ports
}

set porthelp(deactivate) {
Deactivates the given ports
}

set porthelp(dependents) {
Returns a list of installed dependents for each of the given ports

Note: Don't get fooled by the language!
Dependents are those ports which depend on the given port, not vice-versa!
}

set porthelp(deps) {
This action is an alias for 'info --pretty --fullname --depends'
}

set porthelp(destroot) {
Destroot the given ports
}

set porthelp(dir) {
Returns the directories of the given ports

This can be quite handy to be used in your shell:
cd $(port dir <portname>)
}

set porthelp(distcheck) {
Checks if the given ports can be fetched from all of its master_sites
}

set porthelp(distfiles) {
Returns a list of distfiles for the given port
}

set porthelp(dmg) {
Creates a dmg for each of the given ports
}

set porthelp(dpkg) {
Creates a dpkg for each of the given ports
}

set porthelp(echo) {
Returns the list of ports the argument expands to

This can be useful to see what a pseudo-port expression expands to.
}

set porthelp(edit) {
Edit given ports
}

set porthelp(ed) $porthelp(edit)

set porthelp(exit) {
Exit port

Only in interactive mode.
}

set porthelp(extract) {
Extract the downloaded files of the given ports
}

set porthelp(fetch) {
Downloaded distfiles for the given ports
}

set porthelp(file) {
Returns the path to the Portfile for each of the given ports
}

set porthelp(gohome) {
Opens the homepages of the given ports in your browser
}

set porthelp(help) {
Displays short help texts for the given commands
}

set porthelp(info) {
Returns information about the given ports. Most of the options specify a field
to be included in the resulting report. Multiple fields may be specified, in
which case all will be included.  If no fields are specified, a useful default
set will be used.  The other options which do not correspond to fields are:
   --depends   An abbreviation for all depends_* fields
   --index     Do not read the Portfile, instead rely solely on the index
               information. Note this option will prevent the information
               reported from reflecting the effects of any variants specified.
   --line      Report on each port on a single line, with fields separated
               by spaces.  Handy for automatically processing the output of
               info called on a large number of ports.
   --pretty    Format the output in a convenient, human-readable fashion. Note
               that this option is the default when no options are specified to
               info.
}

set porthelp(install) {
Installs the given ports
}

set porthelp(installed) {
List installed versions of the given port, or all installed ports if no port is given
}

set porthelp(lint) {
Checks if the Portfile is lint-free for each of the given ports
}

set porthelp(list) {
List the available version for each of the given ports
}

set porthelp(livecheck) {
Checks if a new version of the software is available
}

set porthelp(load) {
Interface to launchctl(1) for ports providing startup items
}

set porthelp(location) {
Returns the install location for each of the given ports
}

set porthelp(mdmg) {
Creates a dmg containing an mpkg for each of the given ports and their dependencies
}

set porthelp(mirror) {
Fetches distfiles for the given ports
}

set porthelp(mpkg) {
Creates an mpkg for each of the given ports and their dependencies
}

set porthelp(outdated) {
Returns a list of outdated ports
}

set porthelp(patch) {
Applies patches to each of the given ports
}

set porthelp(pkg) {
Creates a pkg for each of the given ports
}

set porthelp(platform) {
Returns the current platform that port is running on
}

set porthelp(provides) {
Return which port provides each of the files given
}

set porthelp(quit) $porthelp(exit)

set porthelp(rpm) {
Creates a rpm for each of the given ports
}

set porthelp(search) {
Search for a port

--case-sensitive   match the search string in a case-sensitive manner
--exact   match the literal search string exactly
--glob    treat the given search string as a glob (default)
--line    print each result on a single line
--regex   treat the given search string as a regular expression
--<field> match against <field>, default is '--name --description'
}

set porthelp(select) {
Select between multiple versions of a versioned port

This allows you to choose which version, among several installed versions
of a port, is to be considered primary.  What this means is which version
becomes the one most would consider the default, e.g. the one run without
specifying any version.

One example is the set of python ports, where there are (among others)
python25, python26, and python30.  The select command lets you set which
of these becomes the version run when you simply use 'python'.

<arguments> must include the group upon which to be acted, and may include
a version if --set is used.

--list   List available versions for the group
--set    Select the given version for the group
--show   Show which version is currently selected for the group (default if
         none given)
}

set porthelp(selfupdate) {
Upgrade MacPorts itself and run the sync target
}

set porthelp(srpm) {
Creates a srpm for each of the given ports
}

set porthelp(submit) {
Submit a port to the MacPorts Web Application (unimplemented)
}

set porthelp(sync) {
Synchronize the set of Portfiles
}

set porthelp(test) {
Run tests on each of the given ports
}

set porthelp(unarchive) {
Unarchive the given ports
}

set porthelp(uninstall) {
Uninstall the given ports
}

set porthelp(unload) $porthelp(load)

set porthelp(upgrade) {
Upgrades the given ports to the latest version. Respects global options
-n, -R, and -u (see the port man page).  Note that in selecting variants
to use in the upgraded build of the port, the order of precedence is variants
specified on the command line, then variants active in the latest installed
version of the port, then the variants.conf file.

--force             Ignore circumstances that would normally cause ports to be
                    skipped (e.g. not outdated).
--enforce-variants  If the installed variants do not match those requested,
                    upgrade even if the port is not outdated.
--no-replace        Do not replace one port with another according to the
                    replaced_by field
}

set porthelp(url) {
Returns the URL for each of the given ports
}

set porthelp(usage) {
Returns basic usage of the port command
}

set porthelp(variants) {
Returns a list of variants provided by the given ports, with descriptions if present
}

set porthelp(version) {
Returns the version of MacPorts
}
